runhouse/servers/http/http_server.py

```diff
@@ -0,0 +1,390 @@
+                                 media_type='application/json')
+
+    def _get_object_and_logs_generator(self, key, stream_logs=False):
+        logfiles = None
+        open_files = None
+        ret_obj = None
+        err = None
+        tb = None
+        returned = False
+        while not returned:
+            try:
+                ret_obj = obj_store.get(key, timeout=self.LOGGING_WAIT_TIME)
+                logger.info(f"Got object of type {type(ret_obj)} back from object store")
+                returned = True
+                # Don't return yet, go through the loop once more to get any remaining log lines
+            except ray.exceptions.GetTimeoutError:
+                pass
+            except ray.exceptions.TaskCancelledError as e:
+                logger.info(f"Attempted to get task {key} that was cancelled.")
+                returned = True
+                ret_obj = None
+                err = e
+                tb = traceback.format_exc()
+            except Exception as e:
+                yield json.dumps(jsonable_encoder(Response(
+                    error=pickle_b64(e),
+                    traceback=pickle_b64(traceback.format_exc()),
+                    output_type=OutputType.EXCEPTION)))
+
+            if stream_logs:
+                if not logfiles:
+                    logfiles = obj_store.get_logfiles(key)
+                    open_files = [open(i, "r") for i in logfiles]
+                    logger.info(f"Streaming logs for {key} from {logfiles}")
+
+                # Grab all the lines written to all the log files since the last time we checked
+                ret_lines = []
+                for i, f in enumerate(open_files):
+                    file_lines = f.readlines()
+                    if file_lines:
+                        # TODO [DG] handle .out vs .err, and multiple workers
+                        # if len(logfiles) > 1:
+                        #     ret_lines.append(f"Process {i}:")
+                        ret_lines += file_lines
+                if ret_lines:
+                    yield json.dumps(jsonable_encoder(Response(
+                        data=ret_lines,
+                        output_type=OutputType.STDOUT,
+                    )))
+
+        if stream_logs:
+            # We got the object back from the object store, so we're done (but we went through the loop once
+            # more to get any remaining log lines)
+            [f.close() for f in open_files]
+        if ret_obj:
+            yield json.dumps(jsonable_encoder(Response(
+                data=codecs.encode(ret_obj, "base64").decode(),
+                output_type=OutputType.RESULT)))
+        else:
+            yield json.dumps(jsonable_encoder(Response(
+                error=pickle_b64(err),
+                traceback=pickle_b64(tb),
+                output_type=OutputType.EXCEPTION)))
+
+    def PutObject(self, request, context):
```

Comment:
can we remove context from all these? i think may have been leftover from grpc?

Reply:
Done!